src/client/scripts/views/SignedIn/DiscoverView.js

```diff
@@ -176,6 +266,88 @@ export class DiscoverView {
             //     ['cities', true],
             //     etc.
             // ]);
+
+            // Variables hosting values for each checkbox
+            // for filter
+            const femaleBox = elm.querySelector('#femaleBox');
+            const maleBox = elm.querySelector('#maleBox');
+            const nonbinaryBox = elm.querySelector('#nonbinaryBox');
+            const messyBox = elm.querySelector('#messyBox');
+            const cleanBox = elm.querySelector('#cleanBox');
+            const veryCleanBox = elm.querySelector('#veryCleanBox');
+            const morningBox = elm.querySelector('#morningRiserBox');
+            const afternoonBox = elm.querySelector('#afternoonRiserBox');
+            const eveningBox = elm.querySelector('#eveningRiserBox');
+            const lowNoiseBox = elm.querySelector('#lowNoiseBox');
+            const medNoiseBox = elm.querySelector('#mediumNoiseBox');
+            const highNoiseBox = elm.querySelector('#loudNoiseBox');
+            const lessGuestsBox = elm.querySelector('#noToLessGuestsBox');
+            const medGuestsBox = elm.querySelector('#someGuestsBox');
+            const highGuestsBox = elm.querySelector('#anyNumberOfGuesBox');
+            const undergradBox = elm.querySelector('#undergradBox');
+            const postgradBox = elm.querySelector('#postgradBox');
+            const otherBox = elm.querySelector('#otherBox');
+
+            // for preferences
+            const minRentBox = elm.querySelector('#minimumRentBox');
+            const maxRentBox = elm.querySelector('#maximumRentBox');
+            const minOccupantsBox = elm.querySelector('#minimumOccupantBox');
+            const maxOccupantsBox = elm.querySelector('#maximumOccupantBox');
+            const leaseSemesterBox = elm.querySelector('#semesterBox');
+            const leaseMonthBox = elm.querySelector('#monthBox');
+            const leaseHalfYearBox = elm.querySelector('#halfYearBox');
+            const leaseFullYearBox = elm.querySelector('#fullYearBox');
+
+
+            // initial map, given all elements are false, currently contains all ids
+            const filterMap = new Map([
+                // items from user
+                // items for gender
+                ['femaleBox', false], // gender.identity === female
+                ['maleBox', false], // gender.identity === male
+                ['nonbinaryBox', false], // gender.identity === nonbinary
```

Comment:
You can use the `checked` property to grab the values in the map itself. This way you can also get rid of the long list of declarations.
```
['femaleBox', elm.querySelector('#femaleBox').checked],
['maleBox', elm.querySelector('#maleBox').checked],
etc.
```

Reply:
Oh I was confused about that - lemme fix it